packages/react-storage/src/components/StorageBrowser/views/hooks/useSearch.ts

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import debounce from 'lodash/debounce';
+
+const DEFAULT_DELAY = 300;
+export interface InitialSearchValues<T> {
+  searchKey: keyof T;
```

Comment:
Not quite sure what `searchKey` is used for or why it's required? Also why `keyof`, afaict from the unit tests, this could just be `string`?

Reply:
`searchKey` is the key used for filtering. e.g LocationsView will provide `prefix` to key into a `LocationData` object to find a value to filter on, and LocationDetail will pass `key` as a searchKey.


`keyof T` is more accurate than using `string` because we want to restrict it to keys in the provided type i.e. LocationData or LocationItem.

example

```
useLocationsView.ts

// this will filter items using the `prefix` key in LocationData. 
// we can make sure the `searchKey` is not an arbitrary string by limiting it to using keyof T`
const { filteredItems } = useSearch<LocationData>({ initialValues: { searchKey: 'prefix' }})


useLocationDetailsView.ts

const { filteredItems } = useSearch<LocationItem>({ initialValues: { searchKey: 'key' }})

```